U-Boot: Add support to board ARV7519RW aka Livebox 2.1

Mandrake-Lee opened the pull request:
Package uboot-lantiq; added support for board ARV7519RW22-A-LT aka Livebox 2.1

The sources have been taken from https://github.com/danielschwierzeck/u-boot-lantiq with original author kept in the patch file.

A couple of important changes:

- Comment variable CONFIG_SYS_DRAM_PROBE in file "arv4579rw.h"; otherwise only half of the memory of the board will be detected.

- Variable CONFIG_SYS_BOOTM_LEN has been redefined to 16Mb. This was leading to crash with the latest initramfs images of Openwrt, which is a real inconvenient during development

The resulting binaries, both .bin an .asc (for uart boot) have been tested with success.

Mandrake-Lee committed:
uboot-lantiq: Add support board ARV7519RW22-A-LT aka Livebox 2.1

Package uboot-lantiq; added support for board ARV7519RW22-A-LT aka Livebox 2.1

The sources have been taken from https://github.com/danielschwierzeck/u-boot-lantiq with original author kept in the patch file.

Significant changes:

-Files and variables renamed in order to better match the hardware i.e arv7519.h ->arv7519rw22.h
-Variable CONFIG_SYS_DRAM_PROBE has been commented in file "arv7519rw22.h"; otherwise only half of the memory of the board will be detected.
-Variable CONFIG_SYS_BOOTM_LEN has been redefined to 16Mb. This was leading to crash with the latest initramfs images of Openwrt, which is a real inconvenient during development
-Defined kernel_addr in environment

The resulting binaries, both .bin an .asc (uart boot) have been tested with success.

Note for future development according @howl & @mkresin:
Recent versions could go beyond the reserved partition size for uboot, making uboot to break itself by writing env variables over itself. Two solutions could be:
A)Make bigger uboot partition also in openwrt image generation
B)Leave the same size but use compression for the phy firmwares.

Signed-off-by: Jorge Amorós-Argos <joramar76@yahoo.es>

hauke commented:
Please follow: https://openwrt.org/submitting-patches
the commit misses a Sob line and please add a description into the commit comment.

howl commented:
I have tested the @mkresin adaptation some time ago https://github.com/mkresin/u-boot-lantiq It's fully functional.

Mandrake-Lee commented:
> 
> 
> Please follow: https://openwrt.org/submitting-patches
> the commit misses a Sob line and please add a description into the commit comment.

Hi, this is probably my 2nd commit in all my life.

I have changed the subject in order to clearly point the package changed; u-boot in this case.

Also I have fully defined the reference of my board.

Can you explain what is a SOB line though?

Mandrake-Lee commented:
Sorry, wrong button! :(

What is missing?

howl commented:
The commit lacks description, it only have the commit title. When you add the description, the last line should look something like this:

Signed-off-by: Your Name <your@email.address>

That's the SOB line.

Mandrake-Lee commented:
I have amended the commit with requested information.

I hope everything is fine now.

howl commented:
I have to go but, for now I see just two appointments:
+#define	MC_CCR41_VALUE	0x141b42
+#define	MC_CCR42_VALUE	0x141b42
capitalize the b
+#define	MC_CCR41_VALUE	0x141B42
+#define	MC_CCR42_VALUE	0x141B42

To be consistent with the naming rename ARV7519RW and arv7519rw to ARV7519RW22 and arv7519rw22.

There are no definition about the kernel_addr variable that should be 0xB0080000 to work with actual images but have to check if it could be necessary or not, but I think it should be defined in /include/configs/arv7519rw(22).h just after:
	CONFIG_ENV_UPDATE_UBOOT_NOR	\
	"kernel_addr=0xB0080000\0"

howl commented:
Forget to mention that there is an issue with more recent uboot versions and this board, if you could add in the description a note that recent versions could go beyond the reserved partition size for uboot, making uboot to break itself by writing env variables over itself. Two solutions could be make bigger uboot partition also in openwrt image generation, or leave the same size but use compression for the phy firmwares.

Mandrake-Lee commented:
> 
> 
> I have to go but, for now I see just three appointments:
> In ddr_config:
> 
> > +#define	MC_CCR41_VALUE	0x141b42
> > +#define	MC_CCR42_VALUE	0x141b42
> 
> capitalize the b
> 
> ```
> +#define	MC_CCR41_VALUE	0x141B42
> +#define	MC_CCR42_VALUE	0x141B42
> ```
> 
> In general: To be consistent with the naming rename ARV7519RW and arv7519rw to ARV7519RW22 and arv7519rw22. Filenames and strings in code.
> 
> In u-boot variables: There are no definition about the kernel_addr variable that should be 0xB0080000 to work with actual images but have to check if it could be necessary or not, but I think it should be defined in /include/configs/arv7519rw(22).h just after:
> 
> ```
> 	CONFIG_ENV_UPDATE_UBOOT_NOR	\
> 	"kernel_addr=0xB0080000\0"
> ```

I'm doing this couple of amendmendts, no real problem.

However the "kernel_addr" definition is redundant. It must be declared somewhere else because it is already in the first uboot that I could compile.

Do you really want to redefine it? It's not a big issue but sounds strange to me.

howl commented:
It seems to be defined in all the uboots already included, not sure what to think about it. Will check where is really defined that memory position for kernel_addr. Are you sure that it's now your actual saved env variables in the uboot env mtd block defined by yourself previously?

Mafketel commented:
How did you fix this problem?

> Variable CONFIG_SYS_BOOTM_LEN has been redefined to 16Mb. This was leading to crash with the latest initramfs images of Openwrt, which is a real inconvenient during development

I see it is still defined in  include/configs/arv7519rw.h
I had already tried to make a minimum kernel and initfs, down to 3.5mb but it still crashes.
I tried different load addresses that did not help either.


p.s. On my testing platform I have the added disadvantage that there is currently no working way of making a new U-boot that works from the NAND, works fine from uart boot, basically the tpl is not generated and thus it does not boot.
Daniel's 2014.7 NAND tpl version boots but crashes on downloading a file with tftp. The UART to ram version does work.

Mandrake-Lee commented:
> How did you fix this problem?
> 
> > Variable CONFIG_SYS_BOOTM_LEN has been redefined to 16Mb. This was leading to crash with the latest initramfs images of Openwrt, which is a real inconvenient during development

In the default sources of Daniel there was no definition of  CONFIG_SYS_BOOTM_LEN at arv7519rw.h. Therefore it was taking default values of 8Mb.

As you have experienced, loading and booting modern openwrt images was giving me an error during decompression. More in detail I'm using images of 4.5Mb, 10Mb uncompressed according binwalk.

Hence I decided to add that definition into file arv7519rw22.h. You can check the patch. It's pretty straight forward.

Mandrake-Lee commented:
> 
> 
> It seems to be defined in all the uboots already included, not sure what to think about it. Will check where is really defined that memory position for kernel_addr. Are you sure that it's not your actual saved env variables in the uboot env mtd block defined by yourself previously?

I'm pretty sure but I will do a double check this weekend now that I have the patch done with your amendments.

I want to be sure that everything is ok before submitting again.

Mafketel commented:
ok got it, I am having this issue even with that variable set and the way you wrote it down made me think that CONFIG_SYS_BOOTM_LEN was the problem and not the solution.
I had already tried a mini initrdfs+kernel but there might be some other issues happening since whatever I set for compresion in ram disk in menuconfig U-boot tells me it is lzma. This is with the zyxel p2812hnuf1, I know it is different but close enough if it solves the lzma error ;)

Mandrake-Lee commented:
Dears,

- I have implemented all suggestions and also added the variable kernel_addr. As howl pointed out, this is not defined by default; my mistake.
- Both images have been tested and work. It's worth noting that the UART uboot is somehow limited by the hardware mode as the speeds are significantly low. Also the openwrt image I uploaded and tested crashed. Needless to say that the same image with the .bin u-boot ran smooth with no problems. Hereafter the log of the UART mode:
```
U-Boot 2013.10-openwrt4 (Mar 08 2019 - 17:44:49) arv7519rw22
Board: Arcadyan ARV7519RW22-A-LT
SoC:   Lantiq VRX288 v1.1
CPU:   125 MHz
IO:    125 MHz
BUS:   125 MHz
BOOT:  UART
DRAM:  128 MiB
Flash: 32 MiB
Using default environment
In:    serial
Out:   serial
Err:   serial
Net:   ltq-eth
```

And now the equivalent .bin image flashed into another device:
```
U-Boot 2013.10-openwrt4 (Mar 08 2019 - 17:45:37) arv7519rw22
Board: Arcadyan ARV7519RW22-A-LT
SoC:   Lantiq VRX288 v1.2
CPU:   500 MHz
IO:    250 MHz
BUS:   250 MHz
BOOT:  NOR
DRAM:  128 MiB
Flash: 32 MiB
In:    serial
Out:   serial
Err:   serial
Net:   ltq-eth
```
I will include the new code into the commit for next revision

adrianschmutzler commented:
You still have formal issues:
- make "add" after "uboot-lantiq:" lower-case
- wrap commit message correctly

adrianschmutzler commented:
Ping.

adrianschmutzler commented:
Since there has been no response (or any progress) for quite some time, I close this PR now. Feel free to request reopening when there is some new development.
